go-controller/pkg/util/ovs.go

```diff
@@ -125,6 +125,67 @@ type execHelper struct {
 
 var runner *execHelper
 
+type ExecUtilRunSvc interface {
+	RunCmd(cmd kexec.Cmd, cmdPath string, envVars []string, args ...string) (string, string, error)
+	Run(cmdPath string, args ...string) (string, string, error)
+	RunWithEnvVars(cmdPath string, envVars []string, args ...string) (string, string, error)
+}
+
+// ExecUtilRunSvcImplStruct implements the methods defined in the ExecUtilRunSvc interface
+type ExecUtilRunSvcImplStruct struct {
+}
+
+// RunCmd invokes the methods of the Cmd interfaces defined in k8s.io/utils/exec to execute commands
+// Note: the cmdPath and args parameter are used only for logging and is not processed
+func (runsvc *ExecUtilRunSvcImplStruct) RunCmd(cmd kexec.Cmd, cmdPath string, envVars []string, args ...string) (string, string, error) {
+	if cmd == nil {
+		return "", "", fmt.Errorf("cmd object cannot be nil")
+	}
+	if len(envVars) != 0 {
+		klog.V(5).Info("setting cmdSetEnv")
+		cmd.SetEnv(envVars)
+	}
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+
+	cmd.SetStdout(stdout)
+	cmd.SetStderr(stderr)
+
+	counter := atomic.AddUint64(&runCounter, 1)
+	logCmd := fmt.Sprintf("%s %s", cmdPath, strings.Join(args, " "))
+	klog.V(5).Infof("exec(%d): %s", counter, logCmd)
+
+	err := cmd.Run()
+	klog.V(5).Infof("exec(%d): stdout: %q", counter, stdout)
+	klog.V(5).Infof("exec(%d): stderr: %q", counter, stderr)
+	if err != nil {
+		klog.V(5).Infof("exec(%d): err: %v", counter, err)
+	}
+	return strings.Trim(strings.TrimFunc(stdout.String(), unicode.IsSpace), "\" \n"),
+		stderr.String(), err
+}
+
+func (runsvc *ExecUtilRunSvcImplStruct) Run(cmdPath string, args ...string) (string, string, error) {
+	cmd := runner.exec.Command(cmdPath, args...)
+	stdout, stderr, err := runsvc.RunCmd(cmd, cmdPath, []string{}, args...)
+	return stdout, stderr, err
+}
+
+func (runsvc *ExecUtilRunSvcImplStruct) RunWithEnvVars(cmdPath string, envVars []string, args ...string) (string, string, error) {
+	cmd := runner.exec.Command(cmdPath, args...)
+	stdout, stderr, err := runsvc.RunCmd(cmd, cmdPath, envVars, args...)
+	return stdout, stderr, err
+}
+
+var RunCmdExecSvcInst ExecUtilRunSvc
```

Comment:
`RunCmdExecRunner`?

Reply:
thanks for the suggestion

Reply:
renamed as `runCmdExecRunner` in the latest PR pushed